### PR TITLE
doc: improve -netinfo help based on feedback from users and devs

### PR DESCRIPTION
**Pull request by jonatack**
Clarify which networks are displayed by the peer counts table (*reachable* networks; follow-up to #23324) in response to questions received over the past months, and a few other improvements.

### CODE REVIEW

**Comment by brunoerg** [comment]
Concept ACK

**Review by w0xlt — APPROVED** [review]
ACK a4da16f

File: src/bitcoin-cli.cpp [hunk 1]
```diff
@@ -606,8 +606,9 @@ class NetinfoRequestHandler : public BaseRequestHandler
...
         "Arguments:\n"
         + strprintf("1. level (integer 0-%d, optional)  Specify the info level of the peers dashboard (default 0):\n", MAX_DETAIL_LEVEL) +
-        "                                  0 - Connection counts and local addresses\n"
-        "                                  1 - Like 0 but with a peers listing (without address or version columns)\n"
+        "                                  0 - Peer counts for each reachable network as well as for block relay peers\n"
+        "                                      and manual peers, and the list of local addresses and ports\n"
+        "                                  1 - Like 0 but preceded by a peers listing (without address and version columns)\n"
         "                                  2 - Like 1 but with an address column\n"
         "                                  3 - Like 1 but with a version column\n"
         "                                  4 - Like 1 but with both address and version columns\n"
```
Comment (laanwj) [inline src/bitcoin-cli.cpp:614]: Heh we kind of went out of hand with the number of combinations here.

**Comment by laanwj** [comment]
Code review ACK a4da16fbd43ea1fff344f1602f7a5ace3d4a8a95

**Review by kristapsk — APPROVED** [review]
utACK a4da16fbd43ea1fff344f1602f7a5ace3d4a8a95

**Pull request merged**